parsers/mcmaster.js

```diff
@@ -1,283 +1,146 @@
+        // Breaks all classes up, looks at all the details after the Course Code or time in a course
+        var classes = courseText.split(_timeDayPairOrCourseCodeLookAhead);
+
+        if (!course) continue;
+
+        var currentSection = _sectionTypeRegex.exec(courseText);
+
+        for (var j = 0; j < classes.length; j++) {
+            var locationProfPair = _locationProfPair.exec(classes[j]);
+            var time = _timeDayPairRegex.exec(classes[j]);
+            var semester = _semesterPairRegex.exec(classes[j]);
+            var section = _sectionTypeRegex.exec(classes[j]);
+            // Check is the current classes segment is a section if it is update the current Section
+            currentSection = section ? section : currentSection;
+
+            if (!locationProfPair || !time || !semester) continue;
+
+            semester = fixDateFormat(semester[1], semester[2]);
+            var where = locationProfPair[1];
+            var professor = locationProfPair[2];
+            // Changes McMasters MoTuWe format to full weekdays
+            var days = time[1].replace(_dayRegex, function fullWeekday(x){return _weekdays[x] + ",";});
+            var daysArray = days.substring(0, days.length-1).split(",");
+
+            for (var k = 0; k < daysArray.length ; k++) {
+                var startTimeSplit = convertTo24Hour(time[2]).split(":");
+                var startTimeObj = {'hour': parseInt(startTimeSplit[0], 10), 'minute': parseInt(startTimeSplit[1], 10)};
+                var endTimeSplit = convertTo24Hour(time[3]).split(":");
+                var endTimeObj = {'hour': parseInt(endTimeSplit[0], 10), 'minute': parseInt(endTimeSplit[1], 10)};
+                var sem_start = new Date(semester[0]);
+                var sem_end = new Date(semester[1]);
+                var semesterStartObj = { year: sem_start.getFullYear(), month: sem_start.getMonth(), day: sem_start.getDate()};
+                var semesterEndObj = { year: sem_end.getFullYear(), month: sem_end.getMonth(), day: sem_end.getDate()};
+                var courseSplit = course[1].split(" ");
+                timetable.push({
+                    'course_code_faculty' : courseSplit[0],
+                    'course_code_number' : courseSplit[1],
+                    'course_number' : currentSection[1],
+                    'course_name' : course[2],
+                    'semester_start' : semesterStartObj,
+                    'semester_end' : semesterEndObj,
+                    'where' : where,
+                    'professor': professor,
+                    'day' : daysArray[k],
+                    'start_time' : startTimeObj,
+                    'end_time' : endTimeObj,
+                    'class_section' : currentSection[2],
+                    'class_type' : currentSection[3]
+                });
+            }
+        }
+    }
+
+    return timetable;
 }
 
-module.exports.convertToCal = convertToCal;
+fs.readFile(filename, 'utf8', function(err, data) {
```

Comment:
we're gonna have to remove this later but leave here for now (for testing purposes)